### PR TITLE
BP-40 Move run() into a separate func

### PR DESCRIPTION
**Pull request by bogdanprodanj**

### CODE REVIEW

**Comment by leonbiton1** [comment]
@bogdanprodanj can you explain what you doing here? i dont find any task related in BP-40

**Comment by bogdanprodanj** [comment]
> @bogdanprodanj can you explain what you doing here? i dont find any task related in BP-40

I'm making it possible to use `upscale-client` without actually starting it. In case bxgateway is run in static mode, I just need to call for recommended peers

File: upscale.go [hunk 1]
```diff
@@ -34,78 +41,31 @@ type upscale struct {
...
-	}
-
-	log.AddHook(hook)
-	return logLevel.String()
-}
-
 // Init initialise upscale - log, server connection, etc
-func Init(id string) {
+// Use 'Run' to start the client
+func Init(id string) error {
```
Comment (Jeeyb) [inline upscale.go:46]: If you change signature, then you need also update GW code. We have this in our plans ?

File: upscale.go [hunk 2]
```diff
@@ -34,78 +41,31 @@ type upscale struct {
...
 	if !disabled {
 		log.Warn("multiple Init calls")
-		return
+		return nil
 	}
+	disabled = false
+
 	logLevel := initLog()
 	if len(id) != 64 {
 		log.Panicf("invalid id %v. id should be 64 chars in size. Please check intergartion", id)
```
Comment (Jeeyb) [inline upscale.go:55]: If we decide to return an error, why we have panic here ?

File: upscale.go (hunk 1 above)
Comment (bogdanprodanj) [reply upscale.go:46]: yes, I'm waiting to merge this, cut new version and update gateway

**Review by Jeeyb — APPROVED** [review]

File: upscale.go [hunk 3]
```diff
@@ -18,81 +19,41 @@ import (
...
 	if !disabled {
 		log.Warn("multiple Init calls")
-		return
+		return nil
 	}
+	disabled = false
+
 	logLevel := initLog()
 	if len(id) != 64 {
-		log.Panicf("invalid id %v. id should be 64 chars in size. Please check intergartion", id)
```
Comment (leonbiton1) [inline upscale.go:95]: why we do it? we want to accept only geth or gw, now we will accept anything

File: upscale.go [hunk 4]
```diff
@@ -115,14 +76,19 @@ func Init(id string) {
...
 		log.Errorf("failed to initilaise connection with upscale server at %v - %v", server, err)
-		log.Info("upscale is disabled")
-		return
+		return fmt.Errorf("failed to initilaise connection with upscale server at %v: %w", server, err)
 	}
+
 	us.client = protobuf.NewUpscaleClient(conn)
 	us.metadata = metadata.Pairs("id", id, "user", user, "password", password)
 
-	go us.run()
```
Comment (leonbiton1) [inline upscale.go:124]: why moved it? to allow gws that not running with upscale to use recommendation? 
if so why we provide upscale service to gw that not running with upscale? why not just run gw with upscale?

**Pull request merged**